cmd/deploy.go

```diff
@@ -207,8 +210,18 @@ func runDeploy(commandContext *cmdctx.CmdContext) error {
 			}
 
 		} else {
-			//fmt.Println("Docker daemon unavailable, performing remote build...")
-			commandContext.Status("flyctl", cmdctx.SINFO, "Docker daemon unavailable, performing remote build...")
+			if !op.DockerAvailable() {
+				if op.LocalOnly() {
+					return fmt.Errorf("Docker daemon unavailable: Local-only set so cannot use to remote build")
+				}
+				commandContext.Status("flyctl", cmdctx.SINFO, "Docker daemon unavailable: Performing remote build...")
+			} else {
+				if op.RemoteOnly() {
+					commandContext.Status("flyctl", cmdctx.SINFO, "Remote-only set: performing remote build...")
+				} else {
+					commandContext.Status("flyctl", cmdctx.SINFO, "Docker daemon available: Still performing remote build...")
```

Comment:
I think this line is unreachable since docker available and remote only false would use the branch on line 173.

Reply:
I pondered that and considered that it was possible if the remote/local flags lost their exclusion line in the deploy op. The fact that it doesn't flag unreachable leaves me tending to leave it there.